src/YarpProxy.Service/Service/Security/ServerCertificateBinding/SniServerCertificateSelector.cs

```diff
@@ -216,7 +216,7 @@ private bool HasAccessToPrivateKey(X509Certificate2 certificate)
                 // However based on https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.signedcms.computesignature?view=netframework-4.6.2,
                 // it seems that this can result in pin prompt even we set silent to true for some situation.
                 // This could be a potential issue which can potentially result in outage. We choose to access PrivateKey directly for verification.
-                return certificate.PrivateKey != null;
+                return certificate.HasPrivateKey;
```

Comment:
Sugg something like:

```cs
using var key = certificate.GetRSAPrivateKey();
return key != null;
```

This isn't strictly the same as what you have here, and help verify that the current user actually can access the private key, not just that the key exists.

Reply:
Well, you know more about the sdk than I do, go ahead and fix it ;)